modules/apps/object/object-web/src/main/resources/META-INF/resources/js/tests/ObjectDetails/ConfigurationContainer.spec.tsx

```diff
@@ -31,46 +55,110 @@ describe('The ConfigurationContainer component', () => {
 		pluralLabel: {en_US: 'pluralLabel'},
 	};
 
-	describe('renders object entry schedule toggle', () => {
-		const scheduleToggleLabel =
-			'allow-users-to-schedule-a-display-expiration-and-review-date-for-entries';
-
+	const renderConfigurationContainer = (
+		customValues: Partial<ObjectDefinition> = {},
+		props: Partial<React.ComponentProps<typeof ConfigurationContainer>> = {}
+	) => {
 		const {result} = renderHook(useObjectDetailsForm, {
-			initialProps: {initialValues, onSubmit: () => {}},
+			initialProps: {
+				initialValues: {...initialValues, ...customValues},
+				onSubmit: () => {},
+			},
 		});
 
-		const configurationContainer = (isEnableObjectEntrySchedule: boolean) =>
-			render(
-				<ConfigurationContainer
-					hasUpdateObjectDefinitionPermission
-					isEnableObjectEntrySchedule={isEnableObjectEntrySchedule}
-					setValues={result.current.setValues}
-					values={result.current.values}
-				/>
-			);
-
-		it('checked or uncheked according to interactions', () => {
-			configurationContainer(false);
-
-			const scheduleToggle = screen.getByRole('switch', {
-				name: scheduleToggleLabel,
-			});
+		render(
+			<ConfigurationContainer
+				hasUpdateObjectDefinitionPermission
+				isEnableObjectEntrySchedule={false}
+				setValues={result.current.setValues}
+				values={result.current.values}
+				{...props}
+			/>
+		);
+	};
 
-			scheduleToggle.click();
+	it('allows enableIndexSearch toggle to be checked and unchecked when definition is not active', async () => {
+		renderConfigurationContainer({active: false});
 
-			expect(scheduleToggle).toBeChecked();
+		const toggle = screen.getByRole('switch', {
+			name: 'enable-indexed-search',
+		});
 
-			scheduleToggle.click();
+		await userEvent.click(toggle);
+		expect(toggle).toBeChecked();
 
-			expect(scheduleToggle).not.toBeChecked();
-		});
+		await userEvent.click(toggle);
+		expect(toggle).not.toBeChecked();
+	});
+
+	it('disables enableIndexSearch toggle when definition is active', () => {
+		renderConfigurationContainer();
 
-		it('disabled when isEnableObjectEntrySchedule is true', () => {
-			configurationContainer(true);
+		expect(
+			screen.getByRole('switch', {name: 'enable-indexed-search'})
+		).toBeDisabled();
+	});
+
+	it('disables enableObjectEntrySchedule toggle when definition is active and isEnableObjectEntrySchedule is true', () => {
+		renderConfigurationContainer(initialValues, {
+			isEnableObjectEntrySchedule: true,
+		});
 
-			expect(
-				screen.getByRole('switch', {name: scheduleToggleLabel})
-			).toBeDisabled();
+		const scheduleToggle = screen.getByRole('switch', {
+			name: 'allow-users-to-schedule-a-display-expiration-and-review-date-for-entries',
 		});
+
+		expect(scheduleToggle).toBeDisabled();
 	});
+
+	it.each(ALL_TOGGLES.filter((label) => label !== 'enable-indexed-search'))(
+		'"%s" toggle can be checked and unchecked correctly',
```

Comment:
What do you think about using "when the setting is active" here to justify the absence of enable-indexed-search?